### DISCUSSION TITLE
Remove all files from a github repository with commit date greater than 5 days

### DISCUSSION THREAD

**Discussion by Rich0101** [community/community#40046] [Repositories] [answered]
### Select Topic Area

Question

### Body

Is it possible to run a git command from Powershell or a .NET program that would remove all files in a remote repository with a commit date that is more than 5 days older than the current date? If so, does anyone know what this command would look like?

Thanks

**Comment by airtower-luna** [comment, marked as answer]
Git history is generally supposed to be permanent, so there's nothing built-in for this. [Rewriting history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) is possible though, so it would certainly be possible to create such a tool. However you should keep in mind that this would cause *a lot* of trouble if you apply it to a repository shared with anyone else, because with each change they'd have to *manually* accept the new history.
